crypto/kyber/kem_kyber.c

```diff
@@ -3,100 +3,188 @@
 
 #include "../evp_extra/internal.h"
 #include "../fipsmodule/evp/internal.h"
+#include "../fipsmodule/kem/internal.h"
 #include "kem_kyber.h"
 #include "pqcrystals_kyber_ref_common/api.h"
 
-// Note: These methods currently default to using the reference code for Kyber.
-// In a future where AWS-LC has optimized options available, those can be
-// conditionally (or based on compile-time flags) called here, depending on
-// platform support.
+// Legacy KEM drivers for kyber.
+// The reference pqcrystals_kyber* inverts our usual convention of success (1)
+// and fails (0). Define wrappers to handle that.
 
-int kyber512r3_keypair_deterministic(uint8_t *public_key  /* OUT */,
-                                     uint8_t *secret_key  /* OUT */,
-                                     const uint8_t *seed  /* IN */) {
-  return pqcrystals_kyber512_ref_keypair_derand(public_key, secret_key, seed);
+static int kyber512r3_keygen_deterministic(uint8_t *public_key,
+                                           uint8_t *secret_key,
+                                           const uint8_t *seed) {
+  return pqcrystals_kyber512_ref_keypair_derand(public_key, secret_key, seed) == 0;
```

Comment:
Where were we checking the return code of `kyber512r3_keypair_deterministic` before that you're now checking the return code of `pqcrystals_kyber512_ref_keypair_derand` here in `kyber512r3_keygen_deterministic`? Does that now need to handle this function returning our "normal" return code?

Reply:
Ahhhh found it in the deleted files.